java/code/src/com/redhat/rhn/taskomatic/LogUtils.java

```diff
@@ -0,0 +1,105 @@
+     * @return the configured logger
+     */
+    public static Logger configureLogger(Class<?> clazz, String outPath, String errPath) {
+
+        String loggerName = clazz.getName();
+        cleanLogging(loggerName);
+
+        LoggerContext context = (LoggerContext) LogManager.getContext(false);
+        final var config = context.getConfiguration();
+        LoggerConfig loggerConfig = config.getLoggerConfig(loggerName);
+
+        // Carry over the log level changes from the config file to allow getting finer logs than INFO
+        Level level = loggerConfig.getName().equals(loggerName) ? loggerConfig.getLevel() : Level.INFO;
+        config.removeLogger(loggerName);
+        Filter outFilter = ThresholdFilter.createFilter(Level.ERROR, Filter.Result.DENY, Filter.Result.ACCEPT);
+        Appender outAppender = FileAppender.newBuilder().withFileName(outPath)
+                .setName(loggerName + "fileAppender_out")
+                .setLayout(PatternLayout.createDefaultLayout(config))
+                .setFilter(outFilter)
+                .build();
+        outAppender.start();
+
+        Filter errFilter = ThresholdFilter.createFilter(Level.ERROR, Filter.Result.ACCEPT, Filter.Result.DENY);
+        Appender errAppender = FileAppender.newBuilder().withFileName(errPath)
+                .setName(loggerName + "fileAppender_err")
+                .setLayout(PatternLayout.createDefaultLayout(config))
+                .setFilter(errFilter)
+                .build();
+        errAppender.start();
+
+        AppenderRef[] refs = new AppenderRef[]{
+                AppenderRef.createAppenderRef(outAppender.getName(), null, null),
+                AppenderRef.createAppenderRef(errAppender.getName(), null, null)
+        };
+        config.addAppender(outAppender);
+        config.addAppender(errAppender);
+        loggerConfig = LoggerConfig.createLogger(false, level, loggerName, "false",
+                refs, null, config, null);
+        loggerConfig.addAppender(outAppender, level, outFilter);
+        loggerConfig.addAppender(errAppender, Level.ERROR, errFilter);
+        config.addLogger(loggerName, loggerConfig);
+
+        context.updateLoggers();
+        return context.getLogger(clazz);
+    }
+
+    /**
+     * Remove the logger appenders
+     *
+     * @param loggerName the name of logger for which to stop and remove the appenders
+     */
+    public static void cleanLogging(String loggerName) {
+        final var config = ((LoggerContext) LogManager.getContext(false)).getConfiguration();
+        LoggerConfig loggerConfig = config.getLoggerConfig(loggerName);
+        if (loggerConfig.getName().equals(loggerName)) {
+            for (var appender : config.getLoggerConfig(loggerName).getAppenders().values()) {
+                appender.stop();
+                config.getLoggerConfig(loggerName).removeAppender(appender.getName());
+                config.getAppenders().remove(appender.getName());
+            }
+        }
```

Comment:
I'm sure it's right, but just to double check. The logger is named after a class, not an object? I mean if it's called `ClassName` when you get it from the config, not `ClassName@394u23234ke23`, then we're safe here :smile:

Reply:
The logger name is whatever we set and it's equivalent to the name in the XML configuration. In our case we set it to the class name, yes.